### ISSUE TITLE
[Android] Focus TextBox Hidden by SoftInput inside a ScrollViewer

### ISSUE THREAD

**Issue by Soap-141** [unoplatform/uno#11300]
### Current behavior

Currently when using `WindowSoftInputMode = SoftInput.AdjustNothing | SoftInput.StateHidden` on Android even if a control is in a `ScrollViewer`, it's hidden by the soft input.

![Animation_68](https://user-images.githubusercontent.com/57812285/218519852-031c48e5-769c-4127-ad93-d63602d1c735.gif)

### Expected behavior

When using `WindowSoftInputMode = SoftInput.AdjustNothing | SoftInput.StateHidden` on Android with a control is in a `ScrollViewer`, it's not hidden by the soft input like when using `WindowSoftInputMode = SoftInput.AdjustResize | SoftInput.StateHidden` or by default on iOS.

![Animation_69](https://user-images.githubusercontent.com/57812285/218519811-88f29673-8c33-4042-8532-bfc8b27b1f62.gif)

### How to reproduce it (as minimally and precisely as possible)

1. Download [SampleScrollSoftInput.zip](https://github.com/unoplatform/uno/files/10724363/SampleScrollSoftInput.zip)
2. Start the project on iOS
3. Focus a `TextBox`
4. Notice the keyboard doesn't hide the control
5. Start the project on Android
6. Notice the keyboard hide the control

### Workaround

It's possible to use `SoftInput.AdjustResize` but I feel like it can lead to undesired behavior in the application.

### Works on UWP/WinUI

None

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.7.37

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

### IDE version

17.4.4

**Comment by jeromelaban** [comment]
Thanks for the report. Why do you need to use this mode?

**Comment by Soap-141** [comment]
I feel like we were supposed to set the flag to `AdjustNothing` and let Uno scroll the content automatically so we don't get any unwanted behavior from Android. As discussed a long time ago in a private issue. https://github.com/unoplatform/nventive-private/issues/375

**Comment by jeromelaban** [comment]
@Soap-141 @takla21 can you take a look at the same sample app on iOS and make a video on how it behaves? Thanks!

**Comment by takla21** [comment]
Here I tweaked the sample so we make sure as discussed that they should not be scrolling and we can see that the behavior on android & ios are different

![android](https://user-images.githubusercontent.com/17461593/219422925-1cd8cab7-7ab9-4fd3-b255-449fd2b16beb.gif)
![ios](https://user-images.githubusercontent.com/17461593/219422929-5dd0ffb2-c0a9-4232-bd0a-59d32d82ae37.gif)

here's the sample with the modification
[SampleScrollSoftInput_noscroll.zip](https://github.com/unoplatform/uno/files/10757980/SampleScrollSoftInput_noscroll.zip)

**Comment by Soap-141** [comment]
![Animation_70](https://user-images.githubusercontent.com/57812285/219424135-8dbc4fa9-3241-4bf6-8a39-49370f175152.gif)

**Comment by jeromelaban** [comment]
Thanks for the tests. Can you tell me how this was previously not an issue in other apps? It's not the first time we're doing this, right?

**Comment by ajpinedam** [comment]
@Soap-141 the behaviour you're looking for in Android is possible by setting these values `SoftInput.AdjustResize | SoftInput.StateVisible` in the `WindowSoftInputMode`:

```csharp
[Activity(
		MainLauncher = true,
		ConfigurationChanges = global::Uno.UI.ActivityHelper.AllConfigChanges,
		WindowSoftInputMode = SoftInput.AdjustResize | SoftInput.StateVisible
	)]
public class MainActivity : Microsoft.UI.Xaml.ApplicationActivity { }
```

This will give you the same result you're getting on iOS.

I found [this documentation](https://developer.android.com/develop/ui/views/touch-and-input/keyboard-input/visibility) (it could definitely be better) for Android making use of these two sets of values

https://user-images.githubusercontent.com/1900897/220249597-c0e5947b-232f-4c88-9ccf-2746d1470ee0.mp4


Below is the behaviour we see when we do not set any value (default behaviour), and this seems to be wrong. 

https://user-images.githubusercontent.com/1900897/220249457-2e8ba497-313b-4100-8760-7afac27d47a9.mp4

We will update Uno Platform so that the default behaviour is the one seen above (first video).

**Comment by Soap-141** [comment]
@ajpinedam @jeromelaban All right, I was under the impression that Uno was supposed to manage this, but if you guys tell me to use `AdjustResize` we'll use it. I will also update our template.

**Comment by Soap-141** [comment]
@ajpinedam Also I wonder why use `StateVisible` instead of `StateHidden`?

**Comment by jeromelaban** [comment]
@Soap-141 Uno handles most modes, but the default seems incorrect to do what you need. Is the default change doing what you need?

**Comment by Soap-141** [comment]
@jeromelaban Just tested `AdjustResize` in our real application and it doesn't seem to work. May be custom controls and complex layouts are causing this. We will have to investigate more. Also I found that while `AdjustResize` did not work, `AdjustPan` did work but it's not the wanted behavior.

**Comment by juliecantin** [comment]
I also got the same result in my app, the page's content did not shift at all to make it remain visible when the keyboard opened.

**Comment by ajpinedam** [comment]
A [PR was shipped](https://github.com/unoplatform/uno/pull/11605) that could also fix this issue. Could you check with the latest Uno to validate?

cc: @juliecantin  and @Soap-141

**Comment by juliecantin** [comment]
> A [PR was merged](https://github.com/unoplatform/uno/pull/11605) two days ago. This PR could also fix this issue. Could you check with the latest Uno to validate?
> 
> cc: @juliecantin and @Soap-141

Unfortunately, the bug still occurs on my app.

**Comment by Soap-141** [comment]
Fix indeed thanks!

**Comment by Soap-141** [comment]
@jeromelaban Is this fix in latest 4.8? Just tested and doesn't seem to work, I thought it was in.

**Comment by jeromelaban** [comment]
@Soap-141 it has not been backported to 4.8. If you need it, you can create a request.
